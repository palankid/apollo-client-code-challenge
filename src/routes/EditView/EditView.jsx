import React from 'react';

import BookOperationsForm from '../../components/BookOperationsForm';
import Header from '../../components/BookOperationsHeader';

import { useEditBook } from './operations/mutations/editBook';

const EditView = () => {
    const { loading, error, book, handleFinish, handleCancel } = useEditBook();

    return (
        <div className="edit-view">
            <Header titleFragment="Edit" />
            <BookOperationsForm
                isCreateMode={false}
                initialValues={book}
                disabled={Boolean(error)}
                loading={loading}
                onFinish={handleFinish}
                onCancel={handleCancel}
            />
        </div>
    );
}

export default EditView;
