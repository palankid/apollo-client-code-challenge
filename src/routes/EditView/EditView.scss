.edit-view {
    max-width: 1200px;
    min-width: 480px;
    margin: auto;
}
