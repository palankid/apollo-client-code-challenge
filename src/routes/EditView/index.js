import EditView from './EditView';
import './EditView.scss';

export default EditView;
