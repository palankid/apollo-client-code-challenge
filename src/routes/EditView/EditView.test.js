import React from 'react';
import { shallow } from 'enzyme';

import EditView from './EditView';

jest.mock('./operations/mutations/editBook', () => ({
    useEditBook: () => ({
        loading: false,
        error: null,
        book: {},
        handleFinish: jest.fn(),
        handleCancel: jest.fn()
    })
}));

describe('EditView', () => {
    it('should test that component is rendered and not changed unintentionally', () => {
        const wrapper = shallow(
            <EditView />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
