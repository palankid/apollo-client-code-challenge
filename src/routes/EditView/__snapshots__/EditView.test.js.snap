// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditView should test that component is rendered and not changed unintentionally 1`] = `
<div
  className="edit-view"
>
  <BookOperationsHeader
    titleFragment="Edit"
  />
  <BookOperationsForm
    disabled={false}
    initialValues={Object {}}
    isCreateMode={false}
    loading={false}
    onCancel={[MockFunction]}
    onFinish={[MockFunction]}
  />
</div>
`;
